utils/providers.py

```diff
@@ -394,22 +311,22 @@ def _setup_providers(prov_class, validate, check_existing):
         'cloud': {
             'navigate': 'clouds_providers',
             'quad': 'cloud_prov',
-            'list': list_cloud_providers
+            'list': list_providers(BaseProvider.type_mapping['infra'].keys())
```

Comment:
This whole section can be built dynamically.......and actually this whole thing may go away - that's my aim